### ISSUE TITLE
Feature request: Limit items for sale to discovered only.

### ISSUE THREAD

**Issue by awdwyn** [sbtoonz/Trader_2.0#19]
**Is your feature request related to a problem? Please describe.**
I'd like to set up Knarr as a materials vendor, but limit the sales to only items that the player has discovered previously.

**Describe the solution you'd like**
Resources / materials in the sales tab filtered to only display those the player has picked up before.

**Describe alternatives you've considered**
I have tried various combinations of the options available in the config file, but those do not seem to limit materials.

**Additional context**
Attached screenshot of the sell tab as configured, I would like it to only show tin, copper, and iron as I have not yet visited the mountains or beyond...
![image](https://user-images.githubusercontent.com/23404043/227803412-45bfd74c-141f-4686-8a59-7451ca31fc9e.png)

Current configuration screen and trader_config.yaml contents...
![image](https://user-images.githubusercontent.com/23404043/227803458-b8369b73-6ca4-40b5-bb6d-3c34f0138cd5.png)

**Comment by sbtoonz** [comment]
yeah the hide known and sell mats thing was added for @horem specifically for his DoD stuff

I can add another config that is "Only sell known mats"

**Comment by awdwyn** [comment]
That would be fantastic, thank you 😄

**Comment by sbtoonz** [comment]
Reading my logic in the code
```
if (Player.m_localPlayer.m_knownRecipes.Contains(itemData.Key.m_itemData.m_shared.m_name))
                    {
                        AddItemToDisplayList(itemData.Key, itemData.Value.Stack, itemData.Value.Cost, itemData.Value.InvCount, ListRoot!, false);
                    }
```
basically what this says in plain english is

Scan all items
While going through all items
If the item is known by the player in their recipes
Add it to the list of items for sale


That would mean the logic behind that switch SHOULD yield the behaviour that you are after.

Meaning That you (player) can only buy things you have discovered 


Am I misunderstanding your feature request? 

I suppose it could be read as "I wanna sell all items regardless of player knowing them but only sell the materials the player has found" 


 let me know if it is indeed that kind of thing and I can add another logic switch for that setting

**Comment by awdwyn** [comment]
I'm interested in the buy tab, where the player can buy items from Knarr. 

The logic there works for craftable recipes, but it seems to fail for items that cannot be crafted (CopperOre for example).
With "only sell known recipes" enabled and "sell mats when hiding unknown recipes" enabled, I should only see CopperOre in the sales list if I have previously picked one up, yet I can see it prior to that.

As an example:
Brand new character, not picked up any items yet, so nothing discovered.
Sell list is:
`CopperOre:
  cost: 15
  stack: 1
  inventory count: 1
  purchase price: 5
Copper:
  cost: 15
  stack: 1
  inventory count: 1
  purchase price: 5
KnifeCopper:
  cost: 15
  stack: 1
  inventory count: 1
  purchase price: 5
`
At the beginning I should not be able to buy anything, but after mining a copper ore that should then become available. Once I smelt it into copper I should then be able to buy that, and then if I pick up some wood and make a forge I should unlock the recipe for the knife (this works already).

As it is now, copper ore and copper (ingot) are available to purchase as a brand new naked player, and the knife unlocks after the new recipe notification pops up when I craft a forge.

I hope that's clearer for you 😄

**Comment by sbtoonz** [comment]
I can knock this out now that I solved a large bug I was tracking

**Comment by sbtoonz** [comment]
https://github.com/sbtoonz/Trader_2.0/releases/tag/v0.4.6
